crates/electrum/tests/test_electrum.rs

```diff
@@ -0,0 +1,240 @@
+
+    // Mine some blocks.
+    env.mine_blocks(101, None)?;
+    wait_for_block(&env, &client)?;
+    let height = env.bitcoind.client.get_block_count()?;
+    let blocks = (0..=height)
+        .map(|i| env.bitcoind.client.get_block_hash(i))
+        .collect::<Result<Vec<_>, _>>()?;
+
+    // Perform reorg on six blocks.
+    env.reorg(6)?;
+    wait_for_block(&env, &client)?;
+    let reorged_height = env.bitcoind.client.get_block_count()?;
+    let reorged_blocks = (0..=height)
+        .map(|i| env.bitcoind.client.get_block_hash(i))
+        .collect::<Result<Vec<_>, _>>()?;
+
+    assert_eq!(height, reorged_height);
+
+    // Block hashes should not be equal on the six reorged blocks.
+    for (i, (block, reorged_block)) in blocks.iter().zip(reorged_blocks.iter()).enumerate() {
+        match i <= height as usize - 6 {
+            true => assert_eq!(block, reorged_block),
+            false => assert_ne!(block, reorged_block),
+        }
+    }
+
+    Ok(())
+}
+
+/// Ensure that confirmed txs that are reorged become unconfirmed.
+///
+/// 1. Mine 101 blocks.
+/// 2. Mine 11 blocks with a confirmed tx in each.
+/// 3. Perform 11 separate reorgs on each block with a confirmed tx.
+/// 4. Check [`Balance`] after each reorg to ensure unconfirmed amount is correct.
+#[test]
+fn tx_can_become_unconfirmed_after_reorg() -> Result<()> {
+    const REORG_COUNT: usize = 11;
+    const SEND_AMOUNT: Amount = Amount::from_sat(10_000);
+
+    let env = TestEnv::new()?;
+    let client = electrum_client::Client::new(env.electrsd.electrum_url.as_str())?;
+
+    // Setup addresses.
+    let addr_to_mine = env
+        .bitcoind
+        .client
+        .get_new_address(None, None)?
+        .assume_checked();
+    let spk_to_track = ScriptBuf::new_v0_p2wsh(&WScriptHash::all_zeros());
+    let addr_to_track = Address::from_script(&spk_to_track, bdk_chain::bitcoin::Network::Regtest)?;
+
+    // Setup receiver.
+    let (mut recv_chain, _) = LocalChain::from_genesis_hash(env.bitcoind.client.get_block_hash(0)?);
+    let mut recv_graph = IndexedTxGraph::<ConfirmationTimeHeightAnchor, _>::new({
+        let mut recv_index = SpkTxOutIndex::default();
+        recv_index.insert_spk((), spk_to_track.clone());
+        recv_index
+    });
+
+    // Mine some blocks.
+    env.mine_blocks(101, Some(addr_to_mine))?;
+
+    // Create transactions that are tracked by our receiver.
+    for _ in 0..REORG_COUNT {
+        env.send(&addr_to_track, SEND_AMOUNT)?;
+        env.mine_blocks(1, None)?;
+    }
+
+    // Sync up to tip.
+    wait_for_block(&env, &client)?;
+    let ElectrumUpdate {
+        chain_update,
+        relevant_txids,
+    } = client.sync(recv_chain.tip(), [spk_to_track.clone()], None, None, 5)?;
+
+    let missing = relevant_txids.missing_full_txs(recv_graph.graph());
+    let graph_update = relevant_txids.into_confirmation_time_tx_graph(&client, None, missing)?;
+    let _ = recv_chain.apply_update(chain_update);
```

Comment:
```suggestion
    let chain_changeset = recv_chain.apply_update(chain_update).unwrap();
    assert_eq!(chain_changeset.len(), REORG_COUNT);
```

This should be true, but right now `change_changeset` only contains 8 elements. I think it's because we only fetch `CHAIN_SUFFIX_LENGTH` number of blocks (8) from electrum, which adds an extra wrinkle I'm not sure was intended.

If we set `REORG_COUNT` to 8, then this is resolved, and we won't find new anchors added post-reorg.

Reply:
Good catch. Fixed.